### PR TITLE
Basic registration with name

### PR DESCRIPTION
**Pull request by jrmhaig**
Also includes reregistration with known id

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -21,6 +21,14 @@
     t.datetime "updated_at",  null: false
   end
 
+  create_table "deviceorama_hive_attributes", force: :cascade do |t|
```
Comment (jonpwilson) [inline db/schema.rb]: Does it need to be 'deviceorama'... ? Would this be sufficient as hive_attributes?

**Comment by jonpwilson** [comment]
Can we remove the .keep files and add it into the .gitignore within the deviceorama_hive directory?

File: db/schema.rb (hunk 1 above)
Comment (jrmhaig) [inline db/schema.rb]: Only that the gem is (will be) deviceorama_hive. We can discuss this.

**Comment by jrmhaig** [comment]
The .keep files were autogenerated by the `rails plugin new` command but I can remove them as I don't think we need to keep the directory structure in github.

**Pull request merged**